src/benchcab/main.py

```diff
@@ -25,8 +25,8 @@ def parse_and_dispatch(parser):
     # Remove the verbose argument
     _ = args.pop("verbose")
 
-    # We just need to instantiate this with the desired level
-    get_logger(level=log_level)
+    logger = get_logger()
+    set_log_level(logger, log_level)
```

Comment:
Rather than setting the level in the other function `set_log_level` (which is similar to the implementation in `get_logger` after the first call, I think we could pass in `level=log_level` argument to `get_logger` as before and have it return the logger with the correct level in the first time itself. Near:

https://github.com/CABLE-LSM/benchcab/blob/94824588d92727aa396acf0298166dc00eacb341/benchcab/utils/__init__.py#L127


This would probably remove the need for `set_log_level` and keeps the implementation as it is?
Something like this
```py
def get_logger(name="benchcab", level="debug"):
    # Get or create a logger
    logger = logging.getLogger(name)
    # Workaround for native singleton property.
    # NOTE: This will ignore the provided level and give you whatever was first set.

    # Set the level everytime
    level = getattr(logging, level.upper())
    logger.setLevel(level)

    if len(logger.handlers):
        return logger


    ....
```
(Haven't tested, but see [StackOverflow](https://stackoverflow.com/questions/6333916/python-logging-ensure-a-handler-is-added-only-once) for more details

Reply:
Ah I didn't consider setting the log level before the return statement. Will try now.